datahub-frontend/app/client/AuthServiceClient.java

```diff
@@ -259,6 +260,44 @@ public boolean verifyNativeUserCredentials(@Nonnull final String userUrn, @Nonnu
     }
   }
 
+  /**
+   * Call the Auth Service to track an analytics event
+   */
+  public void track(@Nonnull final String event) {
+    Objects.requireNonNull(event, "event must not be null");
+    CloseableHttpClient httpClient = HttpClients.createDefault();
+
+    try {
+      final String protocol = this.metadataServiceUseSsl ? "https" : "http";
+      final HttpPost request = new HttpPost(
+          String.format("%s://%s:%s/%s", protocol, this.metadataServiceHost, this.metadataServicePort,
+              TRACK_ENDPOINT));
+
+      // Build JSON request to track event.
+      request.setEntity(new StringEntity(event));
+
+      // Add authorization header with DataHub frontend system id and secret.
+      request.addHeader(Http.HeaderNames.AUTHORIZATION, this.systemAuthentication.getCredentials());
+
+      CloseableHttpResponse response = httpClient.execute(request);
+      final HttpEntity entity = response.getEntity();
+
+      if (response.getStatusLine().getStatusCode() != HttpStatus.SC_OK || entity == null) {
+        throw new RuntimeException(
+            String.format("Bad response from the Metadata Service: %s %s", response.getStatusLine().toString(),
+                response.getEntity().toString()));
+      }
+    } catch (Exception e) {
+      throw new RuntimeException("Failed to verify credentials for user", e);
+    } finally {
+      try {
+        httpClient.close();
```

Comment:
do we need to open & close this client each time we make a request? theres no re-usable client?

Reply:
Refactored to use one client